packages/web3-eth2-beaconchain/src/index.ts

```diff
@@ -0,0 +1,606 @@
+import initWeb3Provider from 'web3-core-provider';
+import {
+    StateId,
+    Status,
+    AttestationData,
+    AttesterSlashing,
+    Web3EthOptions,
+    ProposerSlashing,
+    SyncCommittee,
+    SignedVoluntaryExit,
+    BeaconBlock,
+    BlockId,
+} from './types';
+import {
+    RpcStringResult,
+    RequestArguments,
+    IWeb3Provider,
+} from 'web3-core-types/lib/types';
+import { ValidTypesEnum } from 'web3-utils/lib/types';
+
+export default class Web3Beacon {
+    private _defaultReturnType: ValidTypesEnum;
+
+    provider: IWeb3Provider;
+
+    constructor(options: Web3EthOptions) {
+        this.provider = initWeb3Provider(options.web3Client);
+        this._defaultReturnType =
+            options.returnType || ValidTypesEnum.PrefixedHexString;
+    }
+
+    /**
+     * Retrieve details of the chain's genesis which can be used to identify chain
+     * @param {object} requestArguments (Optional) rpcOptions, providerOptions, and desired returnType rpcOptions, providerOptions, and desired returnType
+     * @returns {Promise} Genesis object
+     */
+    async getGenesis(
+        requestArguments?: Partial<RequestArguments>
+    ): Promise<RpcStringResult> {
+        try {
+            return await this.provider.request({
+                ...requestArguments,
+                method: 'genesis',
+                ethVersion: 2,
+            });
+        } catch (error) {
+            throw Error(`Error getting genesis: ${error.message}`);
+        }
+    }
+
+    /**
+     * Calculates HashTreeRoot for state with given stateId. If stateId is root, same value will be returned.
+     * @param {StateId} stateId State identifier, can be "head", "genesis", "finalized", "justified", slot, hex string encoded stateRoot with 0x prefix
+     * @param {object} requestArguments (Optional) rpcOptions, providerOptions, and desired returnType rpcOptions, providerOptions, and desired returnType
+     * @returns {Promise} HashTreeRoot
+     */
+
+    async getStateRoot(
+        stateId: StateId,
+        requestArguments?: Partial<RequestArguments>
+    ): Promise<RpcStringResult> {
+        try {
+            return await this.provider.request({
+                ...requestArguments,
+                method: `states/${stateId}/root`,
```

Comment:
these are restful endpoints, so should not be called `method`, as now HTTP and WS providers will not be eip1193 so it should be renamed accordingly in request method. this change will increase maintainability/readability.

Reply:
If we make this change, then we can probably get rid of `ethVersion`, as now we detect via the presence of `endpoint` vs `method` in `Web3ProvidersHttp.request`